src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/BookBinder.java

```diff
@@ -0,0 +1,132 @@
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+
+/**
+* Represents Book Binder, a machine that binds multiple enchantments books into one.
+* @author ProfElements
+*/
+public class BookBinder extends AContainer {
+
+    private final ItemSetting<Boolean> bypassVanillaMaxLevel = new ItemSetting<>("bypass-vanilla-max-level", false);
+    private final ItemSetting<Boolean> hasCustomMaxLevel = new ItemSetting<>("has-custom-max-level", false);
+    private final ItemSetting<Integer> customMaxLevel = new IntRangeSetting("custom-max-level", 0, 15, Integer.MAX_VALUE);
+
+    @ParametersAreNonnullByDefault
+    public BookBinder(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+        
+        addItemSetting(bypassVanillaMaxLevel, hasCustomMaxLevel, customMaxLevel); 
+    }
+
+    @Override
+    protected MachineRecipe findNextRecipe(BlockMenu menu) {
+        for (int slot : getInputSlots()) {
+            ItemStack target = menu.getItemInSlot(slot == getInputSlots()[0] ? getInputSlots()[1] : getInputSlots()[0]);
+
+            if (!isCompatible(target)) {
+                return null;
+            }
+
+            ItemStack item = menu.getItemInSlot(slot);
+            if  (isCompatible(item) && isCompatible(target)) {
+                Map<Enchantment, Integer> enchantments = new HashMap<>();
+                EnchantmentStorageMeta itemMeta = (EnchantmentStorageMeta) item.getItemMeta();
+                EnchantmentStorageMeta targetMeta = (EnchantmentStorageMeta) target.getItemMeta();
+
+                Map<Enchantment, Integer> storedItemEnchantments = itemMeta.getStoredEnchants();
+                Map<Enchantment, Integer> storedTargetEnchantments = targetMeta.getStoredEnchants();
+                enchantments.putAll(storedItemEnchantments);
+
+                for (Map.Entry<Enchantment, Integer> entry : storedTargetEnchantments.entrySet()) {
+                    enchantments.merge(entry.getKey(), entry.getValue(), (a, b) -> {
+                            if (a == b) {
+                                if (hasCustomMaxLevel.getValue()) {
+                                    return a + 1 > customMaxLevel.getValue() ? customMaxLevel.getValue() : a + 1; 
+                                } else {
+                                    return a + 1;
+                                }
+                                
+                            } else {
+                                if (hasCustomMaxLevel.getValue()) {
+                                    return Math.max(a, b) > customMaxLevel.getValue() ? customMaxLevel.getValue() : Math.max(a, b);
+                                } else {
+                                    return Math.max(a, b);
+                                }
+                               
+                            }
+                    });
```

Comment:
This is getting a bit out of hand in terms of nesting :/
You could make a seperate private `combineEnchantments` method which takes two inputs and then just reference it using `this::combineEnchantments`. That would reduce the nesting here and make it a bit cleaner to read.